src/AmastyLabelServiceProvider.php

```diff
@@ -17,11 +19,25 @@ public function boot()
 
     public function bootEventyFilters(): self
     {
-        Eventy::addFilter('product.scopes', fn ($scopes) => array_merge($scopes ?: [], [WithProductAmastyLabelScope::class]));
-        Eventy::addFilter('product.casts', fn ($casts) => array_merge($casts ?: [], ['amasty_label' => CastAmastyLabelVariables::class]));
+        config('rapidez.models.product')::resolveRelationUsing('category_amasty_labels', function (Model $product): HasMany {
+            return $product
+                ->hasMany(AmastyLabel::class, 'product_id', 'entity_id')
+                ->where('amasty_label_catalog_parts.type', 1);
+        });
+
+        config('rapidez.models.product')::resolveRelationUsing('product_amasty_labels', function (Model $product): HasMany {
+            return $product
+                ->hasMany(AmastyLabel::class, 'product_id', 'entity_id')
+                ->where('amasty_label_catalog_parts.type', 2);
+        });
+
+        Eventy::addFilter('index.product.scopes', fn ($scopes) => array_merge($scopes ?: [], [WithAmastyCategoryLabelsScope::class]));
```

Comment:
`productpage.scopes` has been re-introduced as well. It. would be nice to eager load `product_amasty_labels` there
https://github.com/rapidez/mirasvit-label/pull/5/files#diff-bf401004cad5b62b99032a8c2eeea508175a058b243a0153ad2e1f17a0a47e65R22